### PR TITLE
feat: Add option to run Dataflow inside a VPC

### PR DESCRIPTION
**Pull request by manitgupta**
FR for #474 

Enables creation of a Dataflow job inside a subnet inside a VPC via both the UI and the CLI.

### CODE REVIEW

**Review by shreyakhajanchi — APPROVED** [review]
Left minor comments. Overall LGTM.

File: streaming/streaming.go [hunk 1]
```diff
@@ -365,7 +367,13 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
 		inputFilePattern = inputFilePattern + "/"
 	}
 	fmt.Println("Reading files from datastream destination ", inputFilePattern)
-
+	dataflowSubnetwork := ""
+	if dataflowCfg.Network != "" && dataflowCfg.Subnetwork == "" {
```
Comment (bharadwaj-aditya) [inline streaming/streaming.go:371]: nit: Merge these 2 conditions as they both check that Network is set.
Also, this method is becoming pretty large, should consider breaking it into smaller chunks

File: ui/src/app/app.constants.ts [hunk 2]
```diff
@@ -101,4 +101,10 @@ export enum ProgressStatus {
...
   { value: 'google_standard_sql', displayName: 'Google Standard SQL Dialect' },
   { value: 'postgresql', displayName: 'PostgreSQL Dialect' },
-]
\ No newline at end of file
+]
+
+export const Dataflow = {
+  Network: 'network',
+  Subnetwork: 'subnetwork',
+  IsDataflowConfigSet: 'IsDataflowConfigSet',
```
Comment (bharadwaj-aditya) [inline ui/src/app/app.constants.ts:109]: nit: different case for the variables. We should follow a consistent naming structure.

File: ui/src/app/components/prepare-migration/prepare-migration.component.ts [hunk 3]
```diff
@@ -271,6 +282,22 @@ export class PrepareMigrationComponent implements OnInit {
     )
   }
 
+  openDataflowForm() {
+    let dialogRef = this.dialog.open(DataflowFormComponent, {
+      width: '30vw',
+      minWidth: '400px',
```
Comment (bharadwaj-aditya) [inline ui/src/app/components/prepare-migration/prepare-migration.component.ts:288]: are these the same dimensions as other forms ? Ideally this sizing should come from CSS

File: webv2/session/types.go [hunk 4]
```diff
@@ -53,6 +53,8 @@ type SessionState struct {
 	SpannerInstanceID   string
 	Dialect             string
 	Region              string
+	Network 			string
```
[the remark this answers is not in the document]
Comment (manitgupta) [reply webv2/session/types.go:56]: Removed.

File: ui/src/app/components/prepare-migration/prepare-migration.component.ts [hunk 5]
```diff
@@ -230,6 +240,7 @@ export class PrepareMigrationComponent implements OnInit {
     localStorage.removeItem(MigrationDetails.IsSourceConnectionProfileSet)
     localStorage.removeItem(MigrationDetails.IsTargetConnectionProfileSet)
     localStorage.removeItem(MigrationDetails.IsSourceDetailsSet)
+    localStorage.removeItem(Dataflow.IsDataflowConfigSet)
     localStorage.removeItem(MigrationDetails.IsMigrationInProgress)
```
[the remark this answers is not in the document]
Comment (manitgupta) [reply ui/src/app/components/prepare-migration/prepare-migration.component.ts:244]: Done

File: ui/src/app/app.constants.ts (hunk 2 above)
Comment (manitgupta) [reply ui/src/app/app.constants.ts:109]: Updated

File: ui/src/app/components/prepare-migration/prepare-migration.component.ts (hunk 3 above)
Comment (manitgupta) [reply ui/src/app/components/prepare-migration/prepare-migration.component.ts:288]: Yes they are the same. I've followed the same convention as other forms for now, but agree that sizing should come from the CSS.
Will look into it as a follow up change.

File: streaming/streaming.go (hunk 1 above)
Comment (manitgupta) [reply streaming/streaming.go:371]: Done

File: streaming/streaming.go [hunk 6]
```diff
@@ -396,14 +390,40 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
 		fmt.Printf("flexTemplateRequest: %+v\n", req)
 		return fmt.Errorf("unable to launch template: %v", err)
 	}
+	reportOutput(conv, datastreamCfg, respDf, project)
+	return nil
+}
+
+func reportOutput(conv *internal.Conv, datastreamCfg DatastreamCfg, respDf *dataflowpb.LaunchFlexTemplateResponse, project string) {
```
Comment (bharadwaj-aditya) [inline streaming/streaming.go:397]: Suggest we create a more descriptive name. Report output seems very generic.

File: webv2/web.go [hunk 7]
```diff
@@ -1739,6 +1747,8 @@ func createStreamingCfgFile(sessionState *session.SessionState, targetDetails ta
 		DataflowCfg: DataflowCfg{
 			JobName:  "",
 			Location: sessionState.Region,
+			Network: dataflowConfig.Network,
+			Subnetwork: dataflowConfig.Subnetwork,
```
Comment (bharadwaj-aditya) [inline webv2/web.go:1751]: nit: is subnetwork a term often used on GCP ? If not, subnet seems like the more commonly used name for this.

Comment (manitgupta) [reply webv2/web.go:1751]: Yes, subnetwork is often used in GCP. 

Ref - https://cloud.google.com/dataflow/docs/guides/specifying-networks

File: streaming/streaming.go (hunk 6 above)
Comment (manitgupta) [reply streaming/streaming.go]: Renamed to `printDataflowJob`

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Pull request merged**